### PR TITLE
Usbl dev

### PR DESCRIPTION
**Pull request by M1chaelM**
@PenguineZhang can you add testing instructions for this PR? Please write a tutorial in the Wiki, or link to one if it's written already. Thank you!

### CODE REVIEW

**Comment by bsb808** [comment]
Responding for @PenguineZhang - working on the tutorial and should be done very soon.

**Comment by M1chaelM** [comment]
To test this PR, see README.md at:
https://gitlab.nps.edu/frl/dave/-/tree/usbl_dev/usbl_gazebo

**Comment by M1chaelM** [comment]
@bsb808 Any chance you have time to review this?

**Comment by bsb808** [comment]
> @bsb808 Any chance you have time to review this?

Can do.

@PenguineZhang are you allowed to add the wiki or do we need to move it over from gitlab?

**Comment by PenguineZhang** [comment]
@bsb808 I am not allowed to post anything other than review to public site. So It'll be better if you or other folks in NPS do that. The tutorial is already included in the usbl_gazebo directory as README.md file.

**Comment by bsb808** [comment]
Roger - will do.

**Comment by bsb808** [comment]
@M1chaelM The day got away from me.  Will attempt to defend time tomorrow.

**Comment by bsb808** [comment]
New tutorial at https://github.com/Field-Robotics-Lab/dave/wiki/usbl_tutorial

File: sonar_dave_plugins/sonar_dave_uuv/src/gazebo_ros_image_sonar.cpp [hunk 1]
```diff
@@ -49,6 +49,7 @@
 #include <sensor_msgs/image_encodings.h>
 #include <cv_bridge/cv_bridge.h>
 #include <opencv2/core/core.hpp>
+#include <opencv2/imgproc/imgproc_c.h>
```
Comment (bsb808) [inline sonar_dave_plugins/sonar_dave_uuv/src/gazebo_ros_image_sonar.cpp:52]: Why are we making changes to the image sonar?

Comment (PenguineZhang) [reply sonar_dave_plugins/sonar_dave_uuv/src/gazebo_ros_image_sonar.cpp:52]: @bsb808 There is a compiling error. It's fixed by adding imgproc_c.h

**Comment by bsb808** [comment]
Subset of email conversation with @PenguineZhang and @M1chaelM 

1. I have gone through the PR.  Currently it doesn't seem to work - may have been something I did.
2. Peng - Please have a look at the current github/usbl_dev branch and the associated githhub tutorial.
    * If this is a small fix, please recommend how to fix via github or email
    * If it is a larger fix, please create a new gitlab branch to so that you can work on it directly to fix.

File: usbl_gazebo/msg/USBLCommand.msg [hunk 2]
```diff
@@ -0,0 +1,3 @@
+int32 transponderID
+int32 commandID
```
[the remark this answers is not in the document]
Comment (PenguineZhang) [reply usbl_gazebo/msg/USBLCommand.msg:2]: @bsb808 I agree, having comments are helpful. commandID is an enumeration, defined in usblCommandId.hpp (will change to .hh in the future).

File: usbl_gazebo/msg/USBLCommand.msg [hunk 3]
```diff
@@ -0,0 +1,3 @@
...
+string data
```
[the remark this answers is not in the document]
Comment (PenguineZhang) [reply usbl_gazebo/msg/USBLCommand.msg:3]: @bsb808 The data field can be used to transmit any text data. It's not cap sensitive. Comments will be helpful here.

**Comment by PenguineZhang** [comment]
@bsb808 The USBL tutorial looks good to me

**Comment by PenguineZhang** [comment]
@bsb808 There's a small bug in my code. The fix is to add the following line after line 6 in transponderPlugin.cpp.
TransponderPlugin::~TransponderPlugin(){}

**Review by bsb808 — APPROVED** [review]
Accepting as is with next steps in https://github.com/Field-Robotics-Lab/dave/issues/61

**Comment by crvogt** [comment]
@bsb808 Still linked with open issue, so will leave branch intact for the time being.

**Pull request merged**